File: RecyclerViewDemo/app/src/main/java/com/lkb/recyclerviewdemo/RecyclerViewAdapter.java
package com.lkb.recyclerviewdemo;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder> {
    List<SomeData> myList;
    public RecyclerViewAdapter(List<SomeData> list) {
        this.myList = list;
    }

    @NonNull
    @Override
    public RecyclerViewAdapter.MyViewHolder onCreateViewHolder(@NonNull ViewGroup parentView, int viewType) {
        View itemView = LayoutInflater.from(parentView.getContext())
                .inflate(R.layout.recycler_view_layout, parentView, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.title.setText(myList.get(position).getTitle() );
        holder.description.setText(myList.get(position).getDescription() );
    }

    @Override
    public int getItemCount() {
        return myList.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        public TextView title, description;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            title = itemView.findViewById(R.id.title);
            description = itemView.findViewById(R.id.description);
        }
    }
}

File: BaseAndroidProject/app/src/main/java/com/lkb/baseandroidproject/RecyclerViewAdapter.kt
package com.lkb.baseandroidproject

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.bundle_item.view.*

class RecyclerViewAdapter(val width: Int, val height: Int) :
    RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder>() {
    class MyViewHolder(val myView: View) : RecyclerView.ViewHolder(myView)

    var myData = mutableListOf<String>( "whats up ", "what are you doing")

//    init {
//        myData.addAll(myData)
//
//    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val rootView = LayoutInflater.from(parent.context)
            .inflate(R.layout.bundle_item, parent, false)
        rootView.imageView2.layoutParams = ConstraintLayout.LayoutParams((width / 4), (width / 4))
        return MyViewHolder(rootView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        //holder.myView.tvItem.text = myData[position]
    }

    override fun getItemCount(): Int {
        return myData.size
    }


}